src/test/java/macrobase/analysis/contextualoutlier/ContextualOutlierDetectorTest.java

```diff
@@ -163,15 +187,79 @@ else if( i >=40 && i < 79){
             		new ArrayRealVector(c2)));
         }
 
-        
-        contextualDetector.searchContextualOutliers(data, 3);
-        Map<Context,BatchTrainScore.BatchResult> context2Outliers = contextualDetector.getContextualOutliers();
+        Map<Context,OutlierClassifier> context2Outliers = contextualDetector.searchContextualOutliers(data);
         assertEquals(context2Outliers.size(), 1);
         
         for(Context context: context2Outliers.keySet()){
         	System.out.println("Context: " + context.toString());
-        	System.out.println("Number of Outliers: " + context2Outliers.get(context).getOutliers().size());
         }
     
     }
+	
+	
+	@Test
+    public void testContextualGivenOutliers() {
+	   //construct a contextual outlier detector
+	   MacroBaseConf conf = new MacroBaseConf();
+	   conf.set(MacroBaseConf.LOW_METRICS, new ArrayList<>());
+       conf.set(MacroBaseConf.HIGH_METRICS, Arrays.asList("A1"));
+       conf.set(MacroBaseConf.TRANSFORM_TYPE, "MAD");
+       
+       
+       
+	   List<String> contextualDiscreteAttributes = new ArrayList<String>();
+	   contextualDiscreteAttributes.add("C1_Discrete");
+	   List<String> contextualDoubleAttributes = new ArrayList<String>();
+	   
+	   conf.set(MacroBaseConf.CONTEXTUAL_DISCRETE_ATTRIBUTES, contextualDiscreteAttributes);
+	   conf.set(MacroBaseConf.CONTEXTUAL_DOUBLE_ATTRIBUTES, contextualDoubleAttributes);
+	   conf.set(MacroBaseConf.CONTEXTUAL_DENSECONTEXTTAU, 0.4);
+	   conf.set(MacroBaseConf.CONTEXTUAL_NUMINTERVALS, 10);
+	   
+	   conf.set(MacroBaseConf.OUTLIER_STATIC_THRESHOLD, 3.0);
+	   
+	   conf.getEncoder().recordAttributeName(1, "A1");
+	   conf.getEncoder().recordAttributeName(2, "C1_Discrete");
+       ContextualOutlierDetector contextualDetector = new ContextualOutlierDetector(conf);
+		
+		
+		List<Datum> data = new ArrayList<>();
+		List<Datum> inputOutliers = new ArrayList<Datum>();
+        for (int i = 0; i < 100; ++i) {
+            double[] sample = new double[1];
+            sample[0] = i;
+            
+            Integer[] c1 = new Integer[1];
+            if(i < 5){
+            	c1[0] = 1;
+            }
+            else if( i >=5 && i < 50){
+            	c1[0] = 2;
+            }else{
+            	c1[0] = 1;
+            }
+            
+           
+            ArrayList<Integer> contexutalDiscreteIntegers = new ArrayList<Integer>();
+            contexutalDiscreteIntegers.add(c1[0]);
+            
+            Datum datum = new Datum(new ArrayList<>(), new ArrayRealVector(sample),
+            		contexutalDiscreteIntegers,
+            		new ArrayRealVector());
+            data.add(datum);
+            
+            if(i < 3){
+            	inputOutliers.add(datum);
+            }
+        
+        }
+
+    	Map<Context,OutlierClassifier> context2Outliers = contextualDetector.searchContextGivenOutliers(data, inputOutliers);
+        assertEquals(context2Outliers.size(), 1);
```

Comment:
You are right! I was simply verifying the printed output. I've fixed it to check the actual outlier context.